File: lecture0-30/lecture009/transpose.cpp
#include<iostream>
#include<vector>
using namespace std;
void input(vector<vector<int>> &a){
    for(int i=0 ; i<a.size() ; i++){
        for(int j=0 ; j<a[0].size() ; j++){
            cin>>a[i][j];
        }
    }
}
void transpose(vector<vector<int>> &arr){
    for(int i=0 ; i<arr.size() ; i++){
        cout<<endl;
        for(int j=0 ; j<arr[0].size() ; j++){
            // arr[i][j]=arr[j][i];           
            cout<<arr[j][i]<<" ";
    }
}
}
// void print(vector<vector<int>> &a){
//     for(int i=0 ; i<a.size(); i++){
//         cout<<endl;
//         for(int j=0 ; j<a[0].size() ; j++){
//             cout<<a[i][j]<<" ";
//         }
//     }
// }
int main(){
    int row , col ;
    cout<<"enter row and col value "<<endl;
    cin>>row>>col;
    vector<vector<int>> arr(row , vector<int> (col , 0));
    cout<<"enter array "<<endl;
    input(arr);
    transpose(arr);
   // print(arr);
}

File: lecture009/transpose.java
import java.util.Scanner ;
public class transpose{
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args){
        int row , col ;
        System.out.println("enter row and col val ");
        row = scn.nextInt();
        col = scn.nextInt();
        int [][] arr = new int [row][col];
        System.out.println("enter array ");
        input(arr);
        System.out.print("array after transpose ");
        transpose(arr);
    }
    public static void input(int [][] arr){
        for(int i=0 ; i<arr.length ; i++){
            for(int j=0 ; j<arr[0].length ; j++){
                arr[i][j]= scn.nextInt();
            }
        }
    }
    public static void transpose(int [][] arr){
        for(int i=0 ; i<arr.length ; i++){
            System.out.println();
            for(int j=0 ; j<arr[0].length ; j++){
                System.out.print(arr[j][i] + " ");
            }
        }
    }
}